app/models/user.rb

```diff
@@ -1,10 +1,30 @@
 class User < ApplicationRecord
-  has_many :tasks
+  before_destroy :destroy_action
+  before_update :update_action
+  has_many :tasks, dependent: :destroy # "dependent: :destroy" はレコード削除時に関連すtasksも削除する(callbackあり)
   validates :name, presence: true, length: { maximum: 30 }
   validates :email, presence: true, length: { maximum: 255 },
                     format: { with: /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i }
   
   before_validation { email.downcase! }
   has_secure_password
   validates :password, presence: true, length: { minimum: 6 }
+
+  private
+
+  def destroy_action
```

Comment:
実施内容が分かるようなメソッド名をつけてください。
```suggestion
  # 例
  def abort_when_destroy_last_admin
```